### ISSUE TITLE
Dumping symbol definitions from file is taking too long

### ISSUE THREAD

**Issue by devoteemous** [Andersbakken/rtags#1313]
I am trying to get all symbol definitions in my codebase. To start with, I have a list of files that are already indexed by rtags. In a loop of files, I dump the symbol definitions of the file. For about 4000 files, average time to dump symbol definitions per file is 12 seconds. Average number of symbol definitions per file is 26. Projecting to 100,000 files, it would take more than 10 days to dump all symbol definitions. I believe this is too slow considering that the whole initial indexing took about 12 hours. 

Is there a way to make this process faster?

**Comment by Andersbakken** [comment]
How do you dump the symbol definitions?

**Comment by devoteemous** [comment]
I use the following command: 
PATH_TO_RC --socket-file PATH_TO_RDM_SOCKET_FILE --path-filter=PATH_TO_SOURCE_FILE -F --display-name --no-context --definition-only

For example:
/usr/bin/rc --socket-file /my_project/rdm_socket --path-filter=/ssd/sb/foo.cpp -F --display-name --no-context --definition-only

**Comment by Andersbakken** [comment]
I'm so sorry for forgetting all about this (and other) bug reports. RTags landed on the backburner for me for a while. 

I'm not sure if you're interested in this issue anymore at all but I get considerably faster times when working on rtags itself. 

After indexing rtags I do this:

time rc --sources | grep "^  /" | sed -e 's,:$,,' | while read i; do rc --path-filter=$i -F"" --definition-only --display-name --no-context > /dev/null; done

real    0m1.396s
user    0m0.722s
sys     0m0.571s

In other words 20ms per file. It's of course possible that your project has way more symbols per file but something seems to be off here. Which version of rtags are you on? I know I made some improvements to the path filter at some points but I wanna say that was several years ago.

Once again. Sorry about the delay.

**Comment by Andersbakken** [comment]
Closing due to inactivity.
